lightning/backend/local.go

```diff
@@ -365,14 +367,22 @@ func (local *local) getImportClient(ctx context.Context, peer *metapb.Peer) (sst
 func (local *local) WriteToTiKV(
 	ctx context.Context,
 	engineFile *LocalFile,
-	region *split.RegionInfo) ([]*sst.SSTMeta, error) {
+	region *split.RegionInfo,
+	start, end []byte,
+) ([]*sst.SSTMeta, *Range, error) {
 	var startKey, endKey []byte
 	if len(region.Region.StartKey) > 0 {
 		_, startKey, _ = codec.DecodeBytes(region.Region.StartKey, []byte{})
 	}
+	if bytes.Compare(startKey, start) < 0 {
+		startKey = start
+	}
 	if len(region.Region.EndKey) > 0 {
 		_, endKey, _ = codec.DecodeBytes(region.Region.EndKey, []byte{})
 	}
+	if bytes.Compare(endKey, end) > 0 {
+		endKey = end
+	}
```

Comment:
is it possible that `endKey` is the empty slice?

Reply:
seems impossible, because we always split regions by each startKey and endKey of the ranges. And I update the check by using the `beforeEnd` method to make it more clear